### PR TITLE
Redesign the Reconnect terminal modal [WD-2772]

### PR DESCRIPTION
**Pull request by lorumic**
## Done

- Redesigned the "Reconnect terminal" modal.

Fixes [WD-2772](https://warthogs.atlassian.net/browse/WD-2772)

## QA

- Perform the following QA steps:
  - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
  - With a local copy of this branch, run as described in the [Readme](https://github.com/canonical/lxd-ui#setting-up-for-development)

1. Browse an instance detail page, [terminal tab](https://lxd-ui-256.demos.haus/ui/default/instances/detail/alex-is-testing/terminal), and click on the "Reconnect" button.
2. Check the new design of the modal.

[WD-2772]: https://warthogs.atlassian.net/browse/WD-2772?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/instances/TerminalPayloadForm.tsx [hunk 1]
```diff
@@ -100,89 +112,72 @@ const TerminalPayloadForm: FC<Props> = ({ close, onFinish }) => {
...
-          <Col size={8}>
-            <Button onClick={addEnvironmentRow} type="button" hasIcon>
-              <Icon name="plus" />
-              <span>Add variable</span>
+        <Input type="submit" hidden />
+        <Label className="u-no-margin--bottom">Environment variables</Label>
+        {formik.values.environment.map((_variable, index) => (
+          <div key={index} className="env-variables">
+            <Input
+              type="text"
```
Comment (edlerd) [inline src/pages/instances/TerminalPayloadForm.tsx:149]: This and the other input is missing a unique id. You can use the index to generate one.
We should also add a `label`, and the `labelClassName="u-off-screen"`.

File: src/pages/instances/TerminalPayloadForm.tsx [hunk 2]
```diff
@@ -100,89 +112,72 @@ const TerminalPayloadForm: FC<Props> = ({ close, onFinish }) => {
...
-          </Row>
-        ))}
-        <Row>
-          <Col size={4} />
-          <Col size={8}>
-            <Button onClick={addEnvironmentRow} type="button" hasIcon>
-              <Icon name="plus" />
-              <span>Add variable</span>
+        <Input type="submit" hidden />
+        <Label className="u-no-margin--bottom">Environment variables</Label>
```
Comment (edlerd) [inline src/pages/instances/TerminalPayloadForm.tsx:145]: This shouldn't be a label, because it doesn't refer to any form element.

**Review by edlerd — APPROVED** [review]

**Comment by al-bakalova** [comment]
When all the environment variables are removed, the modal resizes. Could we please prevent it from doing so?
Screenshot 2023-03-28 at 14 59 33

The second key/value pair should only be visible upon request. 
Screenshot 2023-03-28 at 15 00 53

When too many key/value pairs are added, the modal becomes too long and the footer even hides.
Screenshot 2023-03-28 at 15 01 53

Can you please add a content scroll? Please make sure to make the actions (Cancel, Reconnect) sticky and always visible.

**Comment by al-bakalova** [comment]
Looks very good to me! 
One final suggestion -  could you please make sure to always scroll down the content so that the "Add variable" button is visible even after adding a couple of variable key-value pairs?

**Comment by lorumic** [comment]
> Looks very good to me! One final suggestion - could you please make sure to always scroll down the content so that the "Add variable" button is visible even after adding a couple of variable key-value pairs?

Done - now the content should smoothly scroll so that the "Add variable" button stays visible after a new pair is added.

**Comment by al-bakalova** [comment]
@lorumic Looks good to me! Thanks for applying all the changes!

**Comment by lorumic** [comment]
@edlerd Quite a few things have changed in terms of code, so it would be probably better to redo the review before merging?

File: src/pages/instances/TerminalPayloadForm.tsx [hunk 3]
```diff
@@ -79,110 +74,128 @@ const TerminalPayloadForm: FC<Props> = ({ close, onFinish }) => {
     }
   };
 
+  const updateContentHeight = () => {
+    updateMaxHeight("content-wrapper", "p-modal__footer", 64, "max-height");
```
Comment (edlerd) [inline src/pages/instances/TerminalPayloadForm.tsx:78]: Where does the 64 come from?
Is it for the modal padding? The modal has a padding of `1rem` top and bottom, so shouldn't we use `2rem` or read the modal padding interactively?

**Review by edlerd — APPROVED** [review]
LGTM, just curious about the magic 64

File: src/pages/instances/TerminalPayloadForm.tsx (hunk 3 above)
Comment (lorumic) [reply src/pages/instances/TerminalPayloadForm.tsx:78]: Yes, currently this "magic" value was obtained empirically from manual tests at different resolutions, trying to find a `px` amount that allowed to keep the footer always visible without taking too much vertical space around the modal.
But the idea for a future improvement would be to make the `updateMaxHeight` function to receive an `additionalOffset` parameter that is not necessarily in the `px` unit, but rather a full "value+unit" string that you then pass to the `calc` function:
```
  const style = `${targetProperty}: calc(100vh - ${offset}px - ${additionalOffset})`;
```
So if you pass "2rem", and you calculate an "above+below" offset of 20px, you'd have: `calc(100vh - 20px - 2rem)`.

**Pull request merged**